src/services/CertificateGenerationService.ts

```diff
@@ -154,42 +154,37 @@ class CertificateGenerationService {
    * Method to retrieve Test Station details from API
    */
   public async getTestStations() {
-    const config: IInvokeConfig = this.config.getInvokeConfig();
-    const invokeParams: any = {
-      FunctionName: config.functions.testStations.name,
-      InvocationType: "RequestResponse",
-      LogType: "Tail",
-      Payload: JSON.stringify({
-        httpMethod: "GET",
-        path: `/test-stations/`,
-      }),
-    };
-    let testStations: ITestStation[] = [];
-    return this.lambdaClient
-      .invoke(invokeParams)
-      .then(
-        (
-          response: PromiseResult<Lambda.Types.InvocationResponse, AWSError>
-        ) => {
-          const payload: any =
-            this.lambdaClient.validateInvocationResponse(response);
-          testStations = JSON.parse(payload.body);
-
-          if (!testStations || testStations.length === 0) {
-            throw new HTTPError(
-              400,
-              `${ERRORS.LAMBDA_INVOCATION_BAD_DATA} ${JSON.stringify(payload)}.`
-            );
+      const config: IInvokeConfig = this.config.getInvokeConfig();
+      const invokeParams: any = {
+          FunctionName: config.functions.testStations.name,
+          InvocationType: "RequestResponse",
+          LogType: "Tail",
+          Payload: JSON.stringify({
+              httpMethod: "GET",
+              path: `/test-stations/`,
+          }),
+      };
+      let testStations: ITestStation[] = [];
+      let retries = 0;
+
+      while (retries < 3) {
+          try {
+              const response: PromiseResult<Lambda.Types.InvocationResponse, AWSError> = await this.lambdaClient.invoke(invokeParams);
+              const payload: any = this.lambdaClient.validateInvocationResponse(response);
+              const testStationsParsed = JSON.parse(payload.body);
+
+              if (!testStationsParsed || testStationsParsed.length === 0) {
+                  throw new HTTPError(400, `${ERRORS.LAMBDA_INVOCATION_BAD_DATA} ${JSON.stringify(payload)}.`
+                  );
+              }
+              testStations = testStationsParsed;
+              return testStations;
+          } catch (error) {
+              retries++;
+              console.error(`There was an error retrieving the test stations: ${error}`);
```

Comment:
Can you include the attempt number on the error log, will help with support and to identify if the retry is working as expected